### PR TITLE
fix stats init and cpu by room and restructure Memory.stats

### PR DESCRIPTION
**Pull request by Nevrdid**

### CODE REVIEW

**Comment by TooAngel** [comment]
DCBOT: 
This repository is under [democratic collaboration](https://github.com/TooAngel/democratic-collaboration) and will be merged automatically.

The merge decision is based on the outcome of the reviews:
 - `Approve` add the reviewer value (number of commits) to the `metric`
 - `Request changes` substract the reviewer value from the `metric`

 The merge will happen after `metric * 10 days` calculated from the last code change.

Please review the PR to make a good democratic decision.

Summoning some reviewers:

 - @TooAngel: Some warm words would be nice
 - @WhiteHalmos: Some warm words would be nice
 - @mschultheiss83: Some warm words would be nice

**Comment by TooAngel** [comment]
I think we need some more variety in the motivations: https://github.com/TooAngel/democratic-collaboration/blob/10days/src/server.py#L21 :-)

File: src/config_brain_memory.js [hunk 1]
```diff
@@ -41,16 +41,18 @@ brain.prepareMemory = function() {
     console.log('Known constructionSites: ' + Object.keys(constructionSites).length);
   }
 
-  var userName = Memory.username || Game.rooms[Memory.myRooms[0]].controller.owner || 'default';
+  var userName = Memory.username || _.find(Game.spawns, 'owner').owner || 'default';
```
Comment (TooAngel) [inline src/config_brain_memory.js]: Nice, I think we can also remove the `'default'`

**Comment by TooAngel** [comment]
DCBOT: A new review, yeah.

    Votes: 195/285
    Coefficient: 0.684210526316
    Days to merge: 3.15789473684
    Age: 0

**Comment by TooAngel** [comment]
DCBOT: Code update recognized, countdown starts fresh.

**Comment by TooAngel** [comment]
DCBOT: A new review, yeah.

    Votes: 231/285
    Coefficient: 0.810526315789
    Days to merge: 1.89473684211
    Age: 0

**Comment by TooAngel** [comment]
DCBOT: A new review, yeah.

    Votes: 259/285
    Coefficient: 0.908771929825
    Days to merge: 0.912280701754
    Age: 0

**Comment by TooAngel** [comment]
DCBOT: Code update recognized, countdown starts fresh.

**Comment by TooAngel** [comment]
DCBOT: Code update recognized, countdown starts fresh.

**Comment by TooAngel** [comment]
DCBOT: Code update recognized, countdown starts fresh.

**Comment by TooAngel** [comment]
DCBOT: Code update recognized, countdown starts fresh.

File: src/config_brain_memory.js (hunk 1 above)
Comment (TooAngel) [inline src/config_brain_memory.js]: this one

**Comment by TooAngel** [comment]
DCBOT: Code update recognized, countdown starts fresh.

**Comment by TooAngel** [comment]
DCBOT: A new review, yeah.

    Votes: 260/286
    Coefficient: 0.909090909091
    Minutes to merge: 1309
    Age in minutes: 711

**Comment by TooAngel** [comment]
DCBOT: A new review, yeah.

    Votes: 260/286
    Coefficient: 0.909090909091
    Hours to merge: 218
    Age in hours: 119

**Comment by TooAngel** [comment]
DCBOT: A new review, yeah.

    Votes: 260/286
    Coefficient: 0.909090909091
    Hours to merge: 21
    Age in hours: 12

**Comment by TooAngel** [comment]
DCBOT: Code update recognized, countdown starts fresh.

**Comment by TooAngel** [comment]
DCBOT: A new review, yeah.

    Votes: 260/286
    Coefficient: 0.909090909091
    Hours to merge: 8
    Age in hours: 0

**Comment by TooAngel** [comment]
DCBOT: Code update recognized, countdown starts fresh.

**Comment by TooAngel** [comment]
DCBOT: Code update recognized, countdown starts fresh.

**Comment by TooAngel** [comment]
DCBOT: Code update recognized, countdown starts fresh.

**Comment by TooAngel** [comment]
DCBOT: Code update recognized, countdown starts fresh.

**Comment by TooAngel** [comment]
DCBOT: Code update recognized, countdown starts fresh.

**Comment by TooAngel** [comment]
DCBOT: Code update recognized, countdown starts fresh.

File: src/config_brain_stats.js [hunk 2]
```diff
@@ -100,34 +97,35 @@ brain.stats.addRoom = function(roomName) {
...
       queue: room.memory.queue.length
     },
-    cpu: Game.cpu.getUsed()
+    cpu: Game.cpu.getUsed() - previousCpu
   });
 
   if (room.storage) {
     let storage = room.storage;
-    brain.stats.add(roomName, '.storage', {
+    brain.stats.add(roomName, 'storage', {
```
Comment (TooAngel) [inline src/config_brain_stats.js]: You could do `STRUCTURE_STORAGE`

File: src/config_brain_stats.js [hunk 3]
```diff
@@ -100,34 +97,35 @@ brain.stats.addRoom = function(roomName) {
...
       energy: storage.store.energy,
       power: storage.store.power
     });
   }
   if (room.terminal) {
     let terminal = room.terminal;
-    brain.stats.add(roomName, '.terminal', {
+    brain.stats.add(roomName, 'terminal', {
```
Comment (TooAngel) [inline src/config_brain_stats.js]: You could do `STRUCTURE_TERMINAL`

File: src/prototype_room_my.js [hunk 4]
```diff
@@ -545,7 +542,7 @@ Room.prototype.reviveRoom = function() {
     this.log('Increase idiot by subscription token');
     let idiotCreeps = this.find(FIND_HOSTILE_CREEPS, {
       filter: function(object) {
-        return object.owner.username !== 'Invader';
+        return object.owner.username != 'Invader';
```
Comment (TooAngel) [inline src/prototype_room_my.js]: Why?

**Comment by TooAngel** [comment]
DCBOT: A new review, yeah.

    Votes: 260/286
    Coefficient: 0.909090909091
    Hours to merge: 4
    Age in hours: 17

File: src/prototype_room_my.js (hunk 4 above)
Comment (Nevrdid) [inline src/prototype_room_my.js]: oO grunt dev or jsbeautify again i supose

**Comment by TooAngel** [comment]
DCBOT: Code update recognized, countdown starts fresh.

**Comment by TooAngel** [comment]
DCBOT: Code update recognized, countdown starts fresh.

**Comment by TooAngel** [comment]
DCBOT: Code update recognized, countdown starts fresh.

**Comment by TooAngel** [comment]
DCBOT: Code update recognized, countdown starts fresh.

**Comment by TooAngel** [comment]
DCBOT: Code update recognized, countdown starts fresh.

**Comment by TooAngel** [comment]
DCBOT: A new review, yeah.

    Votes: 0/0
    Coefficient: 0
    Hours to merge: -22
    Age in hours: 21

**Comment by TooAngel** [comment]
DCBOT: A new review, yeah.

    Votes: 261/288
    Coefficient: 0.90625
    Hours to merge: 1
    Age in hours: 21

**Pull request merged**